pexec/managed_process_test.go

```diff
@@ -193,6 +209,63 @@ func TestManagedProcessManage(t *testing.T) {
 			test.That(t, err.Error(), test.ShouldContainSubstring, "exit status 1")
 		}
 	})
+	t.Run("OnCrashHandler", func(t *testing.T) {
+		t.Run("returns true and process is restarted", func(t *testing.T) {
+			logger := golog.NewTestLogger(t)
+
+			tempFile := testutils.TempFile(t, "something.txt")
+			defer tempFile.Close()
+
+			watcher, err := fsnotify.NewWatcher()
+			test.That(t, err, test.ShouldBeNil)
+			defer watcher.Close()
+			watcher.Add(tempFile.Name())
+
+			var onCrashHandlerCallCount int
```

Comment:
Make and use this as an atomic, as it's getting incremented into a routine where it COULD be accessed simultaneously. Currently anyway. If you modify to fail after a count of X, and only read after that, then it's probably okay as is.